src/core/kconfigelektra.cpp

```diff
@@ -0,0 +1,337 @@
+    }
+
+    return KConfigKey{
+        group.str(),
+        key_name,
+    };
+}
+
+KConfigBackend::ParseInfo
+KConfigElektra::parseConfig(const QByteArray & /*locale*/, KEntryMap &entryMap, KConfigBackend::ParseOptions options)
+{
+    //TODO error handling
+    //TODO properly handle parse options (or figure out how they actually work)
+    //TODO read meta keys for proper key options
+
+    Key parentKey = Key(read_key(), KEY_END);
+
+    bool oGlobal = options & ParseGlobal;
+
+    this->kdb->get(*this->ks, parentKey);
+
+    for (auto iterator = this->ks->cbegin(); iterator != this->ks->cend(); iterator++) {
+
+        Key key = iterator.get();
+
+        if (!key.isBelowOrSame(parentKey)) {
+            continue;
+        }
+
+        if (key.isDirectBelow(parentKey)) {
+
+            KEntryMap::EntryOptions entryOptions = nullptr;
+
+            if (oGlobal) {
+                entryOptions |= KEntryMap::EntryGlobal;
+            }
+
+            entryMap.setEntry(
+                QByteArray::fromStdString("<default>"),
+                QByteArray::fromStdString(key.getBaseName()),
+                QByteArray::fromStdString(key.getString()),
+                entryOptions
+            );
+        } else {
+
+            KEntryMap::EntryOptions entryOptions = nullptr;
+
+            if (oGlobal) {
+                entryOptions |= KEntryMap::EntryGlobal;
+            }
+
+            auto parentIter = parentKey.begin();
+            auto childIter = key.begin();
+            traverseIterators(&parentIter, &childIter);
+
+            auto childEnd = key.end();
+
+            auto configKey = elektraKeyToKConfigKey(&childIter, &childEnd);
+
+            entryMap.setEntry(
+                QByteArray::fromStdString(configKey.group),
+                QByteArray::fromStdString(configKey.key),
+                QByteArray::fromStdString(key.getString()),
+                entryOptions
+            );
+        }
+    }
+
+    return ParseOk;
+}
+
+inline std::string kConfigGroupToElektraKey(std::string group, const std::string &keyname)
+{
+    if (group == "<default>" || group.empty()) {
+        return keyname;
+    } else {
+        std::replace(group.begin(), group.end(), '\x1d', '/');
+
+        return group.append("/")
+               .append(keyname);
+    }
+}
+
+bool
+KConfigElektra::writeConfig(const QByteArray & /*locale*/, KEntryMap &entryMap, KConfigBackend::WriteOptions options)
+{
+    //TODO merge
+    bool onlyGlobal = options & WriteGlobal;
+
+    const KEntryMapConstIterator end = entryMap.constEnd();
+
+    Key write_key = Key(this->write_key(), KEY_END);
+
+    this->kdb->get(*this->ks, write_key);
+
+    setLocalFilePath(QString::fromStdString(write_key.getString()));
+
+    for (KEntryMapConstIterator it = entryMap.constBegin(); it != end; ++it) {
+        const KEntryKey &entryKey = it.key();
+        KEntry entry = it.value();
+
+        if (entryKey.mKey.isEmpty()) { // ignore group attributes for the moment
```

Comment:
is this a TODO?

Reply:
yes, this is (more or less) part of #10, to write the group attributes to the files again